docs/indexer_architecture.md

```diff
@@ -0,0 +1,19 @@
+# Indexer
+`claircore.internal.indexer`  
+
+The `Indexer` package is responsible for retreiving `Manifest` layers, parsing the contents of each layer, and computing an `IndexReport`.  
+The `Indexer` is implemented as an FSM to correctly report it's current state to the client.  
+
+## States
+The following diagram expresses the possible states of the Indexer  
+![alt text](./indexer_state_diagram.png "indexer controller state diagram")  
```

Comment:
These image tags should be `![indexer controller state diagram](./indexer_state_diagram.png)`